modules/conversion-center/classes/render/core-render.php

```diff
@@ -0,0 +1,291 @@
+			<?php
+			foreach ( $ctas_value as $key => $cta ) {
+				// Bail if no text
+				if ( empty( $cta['cta_link_text'] ) ) {
+					break;
+				}
+				$formatted_link = $this->get_formatted_link_based_on_type_for_cta( $cta );
+
+				// Bail if no link
+				if ( empty( $formatted_link ) ) {
+					break;
+				}
+
+				$ctas_classnames = 'e-link-in-bio__cta is-type-' . $ctas_props_type;
+
+				if ( 'button' === $ctas_props_type && $ctas_props_show_border ) {
+					$ctas_classnames .= ' has-border';
+				}
+
+				if ( 'button' === $ctas_props_type ) {
+					$ctas_classnames .= ' has-corners-' . $ctas_props_corners;
+				}
+
+				$this->widget->add_render_attribute( 'cta-' . $key, [
+					'class' => $ctas_classnames,
+					'href'  => esc_url( $formatted_link ),
+				] );
+
+				if ( 'File Download' === $cta['cta_link_type'] ) {
+					$this->widget->add_render_attribute( 'cta-' . $key, [
+						'download' => 'download',
+					] );
+				}
+				?>
+				<a <?php echo $this->widget->get_render_attribute_string( 'cta-' . $key ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+					<span class="e-link-in-bio__cta-text">
+						<?php echo esc_html( $cta['cta_link_text'] ); ?>
+					</span>
+				</a>
+			<?php } ?>
+		</div>
+		<?php
+	}
+
+	private function render_icons(): void {
+
+		$icons_props_size = $this->settings['icons_size'] ?? 'small';
+		$icons_value      = $this->settings['icon'] ?? [];
+		$has_icons        = ! empty( $icons_value );
+		if ( ! $has_icons ) {
+			return;
+		}
+
+		?>
+		<div class="e-link-in-bio__icons">
+			<?php
+			foreach ( $icons_value as $key => $icon ) {
+				// Bail if no icon
+				if ( empty( $icon['icon_icon'] ) ) {
+					break;
+				}
+
+				$formatted_link = $this->get_formatted_link_for_icon( $icon );
+
+				// Bail if no link
+				if ( empty( $formatted_link ) ) {
+					break;
+				}
+
+				$this->widget->add_render_attribute( 'icon-' . $key, [
+					'class' => 'e-link-in-bio__icon has-size-' . $icons_props_size,
+				] );
+
+				$this->widget->add_render_attribute( 'icon-link-' . $key, [
+					'aria-label' => esc_attr( $icon['icon_platform'] ),
+					'class'      => 'e-link-in-bio__icon-link',
+					'href'       => esc_url( $formatted_link ),
+					'rel'        => 'noopener noreferrer',
+					'target'     => '_blank',
+				] );
+				?>
+				<div <?php echo $this->widget->get_render_attribute_string( 'icon-' . $key ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+					<a <?php echo $this->widget->get_render_attribute_string( 'icon-link-' . $key ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+						<?php Icons_Manager::render_icon( $icon['icon_icon'], [ 'aria-hidden' => 'true' ] ); ?>
+					</a>
+				</div>
+			<?php } ?>
+		</div>
+		<?php
+	}
+
+	private function render_bio(): void {
+		$this->settings = $this->widget->get_settings_for_display();
```

Comment:
@nicoladj77 Why do you still need this function call? AFAIU `$this->settings` already has the settings at this point?

Reply:
@louiswol94 sorry forgot to delete it!